### ISSUE TITLE
“Swift Language Version” (SWIFT_VERSION) build error

### ISSUE THREAD

**Issue by RazvanCosmeanu** [shahen94/react-native-video-processing#60]
`“Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.`

Hello , and thanks for the work you've put in this library, I look forward to using it.
~~Don't really know if this is more of an xcode problem or about plans of supporting the new Swift version.I'm trying to install on xcode 8.3, react-native 0.43.3, and I get the above error. If I go to my Build Settings / Swift Language Version, there's a Swift 2.3 (unsupported) option and a Swift 3 one. 
I get compiler errors on Swift 3. I tried converting the code with Edit/Convert/To Current Swift Syntax... and I get rid of some of the errors, but there's still about 50 of them left. 
I tried my google fu for fixing it, but I'm no swift/objective-c developer
Cheers! 
Edit~~
Update: 

Downgraded version of xcode to 8.21 to use legacy version of swift and I still get some errors:
 ![errors](http://i.imgur.com/DF2grSz.png).

**Comment by shahen94** [comment]
@gevorg94  can you please take a look what the conflict is with Swift version

**Comment by shahen94** [comment]
@RazvanCosmeanu Please take a look this solution.
http://stackoverflow.com/a/41157875

**Comment by RazvanCosmeanu** [comment]
@shahen94 thanks for the tip, but it was being correctly set. 

Maybe there's something wrong in my install flow:

`//package.json`
`"dependencies": {
    "react": "^15.4.0",
    "react-native": "0.43.3",
    "react-native-camera": "git+https://github.com/lwansbrough/react-native-camera.git",
    "react-native-video-processing": "^1.9.0",
    "react-navigation": "git+https://github.com/react-community/react-navigation.git"
  }`

`XCode Version 8.2.1 (8C1002)`

1. npm install --save react-native-video-processing

2. Add files to project {AppName} /node_modules/react-native-video-processing/RNVideoProcessing - create new group, copy files if needed

3. Add files to project {AppName} /node_modules/react-native-video-procesing/GPUImage/framework/GPUImage.xcodeproj - create new group, copy files if needed

4. Add 

- CoreMedia
- CoreVideo
- OpenGLES
- AVFoundation
- QuartzCore
- GPUImage
- MobileCoreServices
to `{AppName}/General/Linked Frameworks and Libraries`

5. Create {AppName}-Bridging-Header.h

`#ifndef <AppName>_Bridging_Header_h`
`#define <AppName>_Bridging_Header_h`

`#import <React/RCTRootView.h>`
`#import <React/RCTView.h>`
`#import <React/RCTBridgeModule.h>`
`#import "RNVideoProcessing.h"`


`#endif /* <AppName>_Bridging_Header_h */`

6. Clean and Run

7. Get
`“Swift Language Version” (SWIFT_VERSION) is required to be configured correctly for targets which use Swift. Use the [Edit > Convert > To Current Swift Syntax…] menu to choose a Swift version or use the Build Settings editor to configure the build setting directly.`

8. Go to `{AppName}/ BuildSettings / Use Legacy Swift Language Version`, change from `Unspecified` to `No`

9. Build and get following errors: 
![error](http://i.imgur.com/YgnJ2dj.png)

10. Go to `{AppName}/ BuildSettings / Use Legacy Swift Language Version`, change from `No`
to `Yes`

11. Build and get 40 more errors

Thanks for the quick responses so far, you're great!

**Comment by shahen94** [comment]
We faced with this issue before.
https://github.com/shahen94/react-native-video-processing/issues/53
Please try to install from master branch directly instead of npm,  I had no time to publish current master branch to npm. It should fix your issue

**Comment by RazvanCosmeanu** [comment]
@shahen94 Installing @1.8.0 made the compile errors go away. Thanks!

**Comment by rossmartin** [comment]
@shahen94 Can you not use this library with Xcode 8.3 and higher?  I'm facing this issue and I cannot downgrade Xcode to 8.21

**Comment by shahen94** [comment]
Hi @rossmartin you should be able to use this lib with XCode 8(2, 3, etc.), Right now we're using XCode 8.3 and this works fine

**Comment by rossmartin** [comment]
@shahen94 Thanks for the quick reply.  It turns out I had to add an entry to the `Objective-C Bridging Header` that points to the `${APPNAME}/Briding-Header.h` file (shown below as future reference for anyone).  I also had to set the `Swift Language Version` to 3.

![image](https://user-images.githubusercontent.com/2498502/28760596-e7498334-756d-11e7-92aa-4c47d656a9df.png)

Thanks for this library - now I'm facing a dreaded `Apple Mach-O Linker (id) Error`.  I must be forgetting to link a binary library or something simple.
